extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibProcessor.java

```diff
@@ -311,6 +313,20 @@ private Containerizer createContainerizer(ContainerImageConfig containerImageCon
         return containerizer;
     }
 
+    private JibContainer containerizeUnderLock(JibContainerBuilder jibContainerBuilder, Containerizer containerizer)
+            throws InterruptedException, RegistryException, IOException, CacheDirectoryCreationException, ExecutionException {
+        Class<?> lockObj = getClass();
```

Comment:
I'm on the fence about falling back to just the current class object, which probably never has an effect.
Alternatively, I could only lock if I actually got the class object from parent CL. WDYT?

Reply:
Thanks @famod for looking into it. Some javadoc explaining the reasons for this change would be good here ;)

Reply:
Yes, of course! I forgot to add a comment myself.

Reply:
I added some javadoc and left the fallback as is.